skrub/_agg_joiner.py

```diff
@@ -145,44 +128,166 @@ class AggJoiner(BaseEstimator, TransformerMixin):
+        main_key=None,
+        aux_key=None,
         cols=None,
         operation=None,
-        suffix=None,
+        suffix="",
     ):
         self.aux_table = aux_table
+        self.key = key
+        self.main_key = main_key
         self.aux_key = aux_key
         self.cols = cols
-        self.main_key = main_key
         self.operation = operation
         self.suffix = suffix
 
+    def _check_dataframes(self, X, aux_table):
+        """Check dataframes input types.
+
+        Raises an error if frames aren't both Pandas or Polars dataframes,
+        or if there is a Polars lazyframe. Alternatively, allows `aux_table` to be "X".
+
+        Parameters
+        ----------
+        X : DataframeLike
+            The main table to augment.
+        aux_table : DataframeLike or "X"
+            The auxiliary table.
+        """
+        if type(aux_table) == str:
+            if aux_table == "X":
+                return X, deepcopy(X)
+            else:
+                raise ValueError("'aux_table' must be a dataframe or 'X'.")
+        # Polars lazyframes will raise an error here.
+        if not hasattr(X, "__dataframe__"):
+            raise TypeError(f"'X' must be a dataframe, got {type(X)}.")
+        if not hasattr(aux_table, "__dataframe__"):
+            raise TypeError(
+                f"'aux_table' must be a dataframe or 'X', got {type(aux_table)}."
+            )
+
+        if (is_pandas(X) and not is_pandas(aux_table)) or (
+            is_polars(X) and not is_polars(aux_table)
+        ):
+            raise TypeError(
+                "'X' and 'aux_table' must be of the same type, got"
+                f"{type(X)} and {type(aux_table)}"
+            )
+
+        return X, aux_table
+
+    def _check_cols(self):
+        """Check `cols` to aggregate.
+
+        If None, `cols` are all columns from `aux_table`, except `aux_key`.
+
+        Returns
+        -------
+        cols
+            1-dimensional array of columns on which to perform aggregation.
+        """
+        cols = atleast_1d_or_none(self.cols)
+        # If no cols provided, all columns but `aux_key` are used.
+        if len(cols) == 0:
+            cols = list(set(self._aux_table.columns) - set(self._aux_key))
+        if not all([col in self._aux_table.columns for col in cols]):
+            raise ValueError("All 'cols' must be present in 'aux_table'.")
+        return cols
+
+    def _check_operation(self):
+        """Check operation input.
+
+        Returns
+        -------
+        operation
+            1-dimensional array of operations to perform on columns.
+        """
+        operation = atleast_1d_or_none(self.operation)
+        if len(operation) == 0:
+            operation = ["mean", "mode"]
+        return operation
+
+    def _check_suffix(self):
+        if not isinstance(self.suffix, str):
+            raise ValueError(f"'suffix' must be a string. Got {self.suffix}")
+
+    def _check_column_name_duplicates_after_aggregation(
```

Comment:
I think we could adapt `_join_utils.check_column_name_duplicates` (maybe we need to add a parameter to indicate if the suffix has already been added or not), and use it to avoid code duplication

Reply:
Yes, let me change that
It could take parameters `main_keys`, `aux_keys` that default to `None` so that it still works the same when not accounting for keys

Reply:
I was debating whether this function is actually very useful for the `AggJoiner`. The aggregations already add suffixes like `_mode` or `_mean`, and in the case of multiple tables in the `MultiAggJoiner` we also add suffixes based on the position of the auxiliary tables in the list. Furthermore, we don't check for duplicate names across tables in the `MultiAggJoiner` since we let every joiner performs the check

To adapt `_join_utils.check_column_name_duplicates`, we'd need to define cols for which it would be okay to overlap (since we could aggregate a table on itself, and thus the key would be the same in `X` and `aux_table`)

Don't know whether to keep it or not @jeromedockes

Reply:
as we need to handle duplicate column names in other estimators as well I'd be ok to leave this topic for a follow-up pr

Reply:
esp since as you say name clashes are less likely in the case of the aggjoiner than eg the Joiner